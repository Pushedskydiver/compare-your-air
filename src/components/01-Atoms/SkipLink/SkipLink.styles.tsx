import styled, { css } from 'styled-components';
import { typography } from '../../../styles/typography.styles';
import { minWidth, maxWidth } from '../../../styles/breakpoints.styles';
import spacing from '../../../styles/spacing.styles';

const SkipLink = styled.a`
  position: absolute;
  top: 0;
  font-weight: ${typography.weight.medium};
  line-height: normal;
  overflow: hidden;
  clip: rect(0 0 0 0);

  &:active,
  &:focus {
    clip: auto;
    overflow: visible;
    z-index: 100;
  }

  ${maxWidth(768, () => css`
    width: 100%;
    margin-right: -5vw;
    margin-left: -5vw;
    padding-top: ${spacing.padding}px;
    padding-bottom: ${spacing.padding}px;
    background-color: var(--foreground-dark);
    color: var(--foreground-light);
    text-align: center;
  `)}

  ${minWidth(768, () => css`
    top: 50%;
    right: 0;
    width: auto;
    transform: translateY(-50%);
  `)}
`;

export default SkipLink;
