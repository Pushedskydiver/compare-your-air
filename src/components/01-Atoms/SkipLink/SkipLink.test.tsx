import React from 'react';
import { cleanup, render } from '@testing-library/react';
import SkipLink from './SkipLink.styles';

afterEach(() => {
  cleanup();
});

describe('<SkipLink />', () => {
  it('should match snapshot', () => {
    const { container } = render(
      <SkipLink href="#main">This is a skip link</SkipLink>
    );

    expect(container).toMatchSnapshot();
  });
})