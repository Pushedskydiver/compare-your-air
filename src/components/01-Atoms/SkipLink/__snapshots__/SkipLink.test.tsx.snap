// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SkipLink /> should match snapshot 1`] = `
<div>
  <a
    class="sc-AxjAm ezykAA"
    href="#main"
  >
    This is a skip link
  </a>
</div>
`;
